diff -urN a/kernel/module/main.c b/kernel/module/main.c
--- a/kernel/module/main.c	2024-12-05 15:09:07.000000000 +0800
+++ b/kernel/module/main.c	2024-12-05 15:14:41.987077500 +0800
@@ -1868,12 +1868,6 @@
 		goto no_exec;
 	}
 
-	if (shdr->sh_size != sizeof(struct module)) {
-		pr_err("module %s: .gnu.linkonce.this_module section size must match the kernel's built struct module size at run time\n",
-		       info->name ?: "(missing .modinfo section or name field)");
-		goto no_exec;
-	}
-
 	info->index.mod = mod_idx;
 
 	/* This is temporary: point mod into copy of data. */
