diff --git a/target/linux/amlogic/meson/base-files/usr/sbin/install-to-emmc.sh b/target/linux/amlogic/meson/base-files/usr/sbin/install-to-emmc.sh
index 5b4c14d..710ab49 100755
--- a/target/linux/amlogic/meson/base-files/usr/sbin/install-to-emmc.sh
+++ b/target/linux/amlogic/meson/base-files/usr/sbin/install-to-emmc.sh
@@ -56,7 +56,8 @@ function part_disk() {
 		mktable msdos \
 		mkpart primary 117MiB 181MiB \
 		mkpart primary 181MiB 580MiB \
-		mkpart primary 800MiB 2848MiB
+		mkpart primary 800MiB 1824MiB \
+		mkpart primary 1824MiB 100%
 }
 
 function get_bootdisk() {
@@ -160,9 +161,13 @@ function main() {
 			dd if=/dev/$partdev of=/dev/$partdevto bs=1M conv=notrunc
 		done
 		# mark RESET overlay partition
-		partdevto=${to}p3
-		echo "mark /dev/$partdevto as RESET"
-		echo "RESET000" | dd of="/dev/$partdevto" bs=512 count=1 conv=sync,fsync
+		for part in 3 4; do
+			partdevto=${to}p$part
+			echo "mark /dev/$partdevto as RESET"
+			echo "RESET000" | dd of="/dev/$partdevto" bs=512 count=1 conv=sync,fsync
+			if grep -qs "/dev/$partdevto" /proc/mounts; then umount -f /dev/$partdevto; fi
+			mkfs.ext4 -F /dev/$partdevto >/dev/null
+		done
 		echo "All Done!"
 	else
 		echo "Bye!"
